### PR TITLE
[Hotfix][CDC] Fix ddl duplicate execution error when config multi_table_sink_replica

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Hotfix] Fix ddl duplicate execution error when config multi_table_sink_replica

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/log4j2-test.properties [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
 # limitations under the License.
-#
-
 ################################################################################
 
-# Set root logger level to OFF to not flood build logs
-# The amount of logs is too large to be suitable for troubleshooting, manually set to WARN.
-rootLogger.level=WARN
-rootLogger.appenderRef.test.ref = TestLogger
+rootLogger.level = INFO
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/log4j2-test.properties]: Please revert.

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/log4j2-test.properties]: fixed

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/sink/writer/IcebergRecordWriter.java [hunk 2]
```diff
@@ -91,6 +91,8 @@ record = recordConverter.convert(seaTunnelRow, rowType);
     public void applySchemaChange(SeaTunnelRowType afterRowType, SchemaChangeEvent event) {
         log.info("Apply schema change start.");
         SchemaChangeWrapper updates = new SchemaChangeWrapper();
+        // get the latest schema in case another process updated it
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/sink/writer/IcebergRecordWriter.java:94]: cc @dailai . So iceberg support schema evolution too? I think we should update the doc.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java [hunk 3]
```diff
@@ -438,72 +441,165 @@ default String convertType(String columnName, String columnType) {
...
      */
-    default void refreshTableSchemaBySchemaChangeEvent(
-            String sourceDialectName,
-            AlterTableColumnEvent event,
-            JdbcConnectionProvider jdbcConnectionProvider,
-            TablePath sinkTablePath) {}
+    default void applySchemaChange(
+            SchemaChangeEvent event, Connection connection, TablePath tablePath)
+            throws SQLException {
+        if (event instanceof AlterTableColumnsEvent) {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java:451]: cc @dailai . Looks like add jdbc supported schema evolution? Not only jdbc mysql.

Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java:451]: Only compatible with MySQL database, other databases need to adapt code logic, add test cases, and documents

**Review by Hisoka-X — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]
LGTM

**Pull request merged**